### PR TITLE
staticanalysis/bot: Fix clang-format bug when no touched/added line in patch

### PR DESCRIPTION
**Pull request by La0**
Fix [this bug](https://tools.taskcluster.net/groups/MQmu293iSrWf8d52Fxr7UQ/tasks/MQmu293iSrWf8d52Fxr7UQ/runs/0/logs/public%2Flogs%2Flive.log)

### CODE REVIEW

**Comment by sylvestre** [comment]
Oh, Python :)

**Comment by marco-c** [comment]
Could you check what is the patch generated by clang-format like here? Maybe we are missing something

**Comment by La0** [comment]
The parsed patch provided by parsepatch is:

```python
# file: netwerk/protocol/http/nsHttpHandler.h
{'added': [], 'deleted': [426], 'touched': [], 'new': False}
```
The raw patch is:

```
diff -r b783cd5203ea netwerk/protocol/http/nsHttpHandler.h
--- a/netwerk/protocol/http/nsHttpHandler.h	Tue Apr 23 08:38:57 2019 +0000
+++ b/netwerk/protocol/http/nsHttpHandler.h	Mon Apr 22 12:41:08 2019 +0300
@@ -418,17 +418,16 @@
   void SetLastActiveTabLoadOptimizationHit(TimeStamp const &when);
   bool IsBeforeLastActiveTabLoadOptimization(TimeStamp const &when);

   bool DumpHpackTables() { return mDumpHpackTables; }

   HttpTrafficAnalyzer *GetHttpTrafficAnalyzer();

   bool GetThroughCaptivePortal() { return mThroughCaptivePortal; }
-
  private:
   nsHttpHandler();

   virtual ~nsHttpHandler();

   MOZ_MUST_USE nsresult Init();

   //
```

**Comment by marco-c** [comment]
Maybe we should consider deleted lines too then?

**Comment by La0** [comment]
Yeah that's why I added @calixteman on this issue: is there a reason the `deleted` lines are ignored ?

File: src/staticanalysis/bot/static_analysis_bot/clang/format.py [hunk 1]
```diff
@@ -104,10 +104,16 @@ def get_eligible_files():
...
+        with open('/tmp/cf.diff', 'w') as f:
+            f.write(reverse_diff)
+
         # Build `ClangFormatIssue`s
         issues = []
         for filename, diff in patch.items():
-            lines = sorted(diff.get('touched', []) + diff.get('added', []))
+            lines = sorted(diff.get('touched', []) + diff.get('added', []) + diff.get('deleted', []))
+            if not lines:
+                logger.warn('No modified lines on this file', file=filename)
```
Comment (marco-c) [inline src/staticanalysis/bot/static_analysis_bot/clang/format.py]: Maybe we should assert here, as if clang-format produces a patch with no modified lines maybe there is an error

Comment (La0) [reply src/staticanalysis/bot/static_analysis_bot/clang/format.py]: :+1:

**Pull request merged**